Makefile

```
@@ -0,0 +1,61 @@
+.PHONY: all check-env docker docker-dev docker-tools dev lint dep build test mock protobuf changelog clean 
+
+all: clean test build docker
+
+MAJOR_VERSION := $(shell echo $(version) | cut -d . -f 1)
+MINOR_VERSION := $(shell echo $(version) | cut -d . -f 1-2)
+PATCH_VERSION := $(version)
+
+check-env:
+ifndef version
+	$(error version is undefined)
+endif
+
+docker: check-env
+	docker build \
+		--build-arg version=$(PATCH_VERSION) \
+		-t mesg/engine:$(MAJOR_VERSION) \
+		-t mesg/engine:$(MINOR_VERSION) \
+		-t mesg/engine:$(PATCH_VERSION) \
+		-t mesg/engine:latest \
+		.
+
+docker-dev:
+	docker build -t mesg/engine:dev --build-arg version=$(version) .
```

Why using `$(version)` without the `check-env`?
I would suggest to hardcode to `dev` the version (like before last commit):
```suggestion
	docker build -t mesg/engine:dev --build-arg version=dev .
```